src/devices/tests/test-acd.c

```diff
@@ -31,6 +33,46 @@
 #define ADDR3 0x03030303
 #define ADDR4 0x04040404
 
+/*****************************************************************************/
+
+static gboolean
+_skip_acd_test_check (void)
+{
+	NAcd *acd;
+	NAcdConfig *config;
+	const guint8 hwaddr[ETH_ALEN] = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06 };
+	int r;
+	static int skip = -1;
+
+	if (skip == -1) {
+		r = n_acd_config_new (&config);
+		g_assert (r == 0);
+
+		n_acd_config_set_ifindex (config, 1);
+		n_acd_config_set_transport (config, N_ACD_TRANSPORT_ETHERNET);
+		n_acd_config_set_mac (config, hwaddr, sizeof (hwaddr));
+
+		r = n_acd_new (&acd, config);
+		n_acd_config_free (config);
+		if (r == 0)
+			n_acd_unref (acd);
+
+		skip = (r != 0);
+	}
+	return skip;
+}
+
+#define _skip_acd_test() \
+	({ \
+		gboolean _skip = _skip_acd_test_check (); \
+		\
+		if (_skip) \
+			g_test_skip ("Cannot create NAcd. Running under valgind?"); \
+		_skip; \
+	})
+
```

Comment:
Wouldn't it be simpler to skip the test from makefile/meson.build when valgrind is enabled?

Reply:
no. The configure option is mostly irrelevant and only affects the default. What matters is whether you enable valgrind when running the test:
```
   NMTST_USE_VALGRIND=1 make -j 5 check
   NMTST_USE_VALGRIND=1 ./tools/run-nm-test.sh ./src/devices/tests/test-acd
   ./tools/run-nm-test.sh -v ./src/devices/tests/test-acd
```
vs.
```
   NMTST_USE_VALGRIND=0 make -j 5 check
   NMTST_USE_VALGRIND=0 ./tools/run-nm-test.sh ./src/devices/tests/test-acd
   ./tools/run-nm-test.sh -V ./src/devices/tests/test-acd
```

Reply:
Understood, thanks.